src/helper_addressGenerator.py

```diff
@@ -0,0 +1,52 @@
+"""
+Create random address
+"""
+
+import time
+import defaults
+import queues
+import state
+
+from bmconfigparser import BMConfigParser
+
+
+def getSortedAccounts():
+    """Get a sorted list of configSections"""
+
+    configSections = BMConfigParser().addresses()
+    configSections.sort(
+        cmp=lambda x, y: cmp(
+            unicode(
+                BMConfigParser().get(
+                    x,
+                    'label'),
+                'utf-8').lower(),
+            unicode(
+                BMConfigParser().get(
+                    y,
+                    'label'),
+                'utf-8').lower()))
+    return configSections
+
+
+def checkHasNormalAddress():
+    """method for checking address"""
+    for address in getSortedAccounts():
+        if BMConfigParser().safeGetBoolean(address, 'enabled'):
+            return address
+    return False
+
+
+def createAddressIfNeeded(label_text, streamNumberForAddress=1):
+    """method for creating random address"""
+    if not checkHasNormalAddress():
+        queues.addressGeneratorQueue.put((
+            'createRandomAddress', 4, streamNumberForAddress,
+            label_text,
+            1, "", False,
+            defaults.networkDefaultProofOfWorkNonceTrialsPerByte,
+            defaults.networkDefaultPayloadLengthExtraBytes
+        ))
+    while state.shutdown == 0 and not checkHasNormalAddress():
```

Comment:
Add another timeout in case something is wrong, so that it doesn't get stuck. In that case just return None or raise an exception.